src/score_hv/harvesters/soca_diags.py

```
@@ -0,0 +1,294 @@
+        self.stats = self.config_data.get('statistic')
+        for stat in self.stats:
+            if stat not in VALID_STATISTICS:
+                msg = ("'%s' is not a supported statistic to harvest from "
+                       "the SOCA diag files.. "
+                       "Please reconfigure the input dictionary using only the "
+                       "following statistics: %r" % (stat, VALID_STATISTICS))
+                raise KeyError(msg)
+
+    def set_depths(self):
+        self.depths = self.config_data.get('depths')
+       
+    def set_regions(self):
+        self.regions = self.config_data.get('regions')
+
+    def get_stats(self):
+        ''' return list of all stat types based on harvest_config '''
+        return self.stats
+
+    def get_variables(self):
+        ''' return list of all variable types based on harvest_config'''
+        return self.variables
+    
+
+@dataclass
+class SOCADiagsHv(object):
+    """ Harvester dataclass used to parse SOCA output.
+    
+        Parameters:
+        -----------
+        config: SOCADiagsConfig object containing information used to determine
+                which variable to parse SOCA output.
+        Methods:
+        --------
+        get_data: parse descriptive statistics from SOCA output.
+    
+                  returns a list of tuples containing specific data
+    """
+    config: SOCADiagsConfig = field(default_factory=SOCADiagsConfig)
+     
+    def get_data(self):
+        depths = self.config.depths
+        groups_wanted = ['ObsValue','oman','ombg']
+        units = None
+        for filename in self.config.harvest_filenames:
+            harvested_data = list()
+            filename_info = parse_filename(filename)
+            print("Opening file: ",filename)
+            try:
+               dataset = netCDF4.Dataset(filename, 'r')
+            except Exception as e: 
+               raise OSError (f"Failed to open NetCDF file {filename}: {e}")
+               sys.exit(1)
```

also would be better to throw the exception for better error handling

---

actually I think it's already doing that with the raise so just need to remove the sys exit here